src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -655,21 +666,34 @@ Coordination::Error ZooKeeper::multiImpl(const Coordination::Requests & requests
         auto response = future_result.get();
         Coordination::Error code = response.error;
         responses = response.responses;
+
+        if (check_session_valid)
+        {
+            if (code != Coordination::Error::ZOK && !Coordination::isHardwareError(code) && getFailedOpIndex(code, responses) == requests.size())
+            {
+                impl->finalize(fmt::format("Session was killed: {}", requests.back()->getPath()));
```

Comment:
do we want to continue and return the user error or do we want to transform this into a hardware error?
I know we don't expect such errors really often, but it seems like an internal check that should be retried.

Reply:
Ah, yes, we should convert it to hardware error